src/robusta/core/model/nodes.py

```diff
@@ -0,0 +1,94 @@
+from pydantic.main import BaseModel
+from typing import List, Dict
+
+
+class NodeInfo(BaseModel):
+    name: str
+    node_creation_time: str
+    internal_ip: str
+    external_ip: str
+    taints: str
+    conditions: str
+    memory_capacity_mb: int
+    allocatable_memory_mb: int
+    allocated_memory_mb: int
+    cpu_capacity: float
+    allocatable_cpu: float
+    allocated_cpu: float
+    pods_count: int
+    pods: str
+    node_info: Dict
+    deleted: bool = False
+
+    def __compare_node_info(self, other_node_info):
+        return self.node_info == other_node_info
+
+    def __eq__(self, other):
```

Comment:
As far as I can tell, you're overriding the Pydantic default just to exclude `deleted` from the comparison. It's currently a little fragile because if someone adds a new field they could forget to add it to __eq__ and cause bugs. 

Could you instead implement __eq__ as follow:

```python
def __eq__(self, other):
  IGNORED_FIELDS = ["deleted"]
  self_no_deleted = {k: v for k,v in self.dict() if k not in ignore_keys} 
  other_no_deleted = {k: v for k,v in other.dict() if k not in ignore_keys} 
  return self_no_deleted == other_no_deleted
```

Reply:
fixed